# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webgpu/ipc/WebGPUParent.cpp
# Commit: dfd34fb1aade
# Full Hash: dfd34fb1aade1295579715b1c21f1e51bbfe370e
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2021-05-18 21:36:28
# Regressor Bug: 1709462
# File Overlap Count: 2
# Description:
#   Bug 1709462 - Fix freeing of WebGPU bind group layouts r=jimb
#   
#   When the pipeline layout is implicit at pipeline creation,
#   we collect the IDs of BGLs, to be able to produce GPUBindGroupLayout
#   object upon user request. However, the produced object didn't know if
# ==============================================================================

diff -r 5c6f0950714d -r dfd34fb1aade dom/webgpu/ipc/WebGPUParent.cpp
--- a/dom/webgpu/ipc/WebGPUParent.cpp	Tue May 18 16:24:49 2021 +0000
+++ b/dom/webgpu/ipc/WebGPUParent.cpp	Tue May 18 16:26:27 2021 +0000
@@ -460,6 +460,15 @@
   return IPC_OK();
 }
 
+ipc::IPCResult WebGPUParent::RecvImplicitLayoutDestroy(
+    RawId aImplicitPlId, const nsTArray<RawId>& aImplicitBglIds) {
+  ffi::wgpu_server_pipeline_layout_drop(mContext, aImplicitPlId);
+  for (const auto& id : aImplicitBglIds) {
+    ffi::wgpu_server_bind_group_layout_drop(mContext, id);
+  }
+  return IPC_OK();
+}
+
 // TODO: proper destruction
 static const uint64_t kBufferAlignment = 0x100;
 